kernel/kernel-api/src/main/java/io/delta/kernel/expressions/Predicate.java

```diff
@@ -0,0 +1,14 @@
+package io.delta.kernel.expressions;
+
+import io.delta.kernel.types.BooleanType;
+import io.delta.kernel.types.DataType;
+
+/**
+ * An {@link Expression} that defines a relation on inputs. Evaluates to true, false, or null.
+ */
+public interface Predicate extends Expression {
```

Comment:
Is this used in any of the interfaces?

Reply:
If not, lets remove this.

Reply:
We in theory could enforce `Predicate` as an input or output type; I don't think we currently have

Reply:
But like we could enforce `Predicates` for `scanBuilder.withFilter` or something